### PR TITLE
plasma mode: Properly spec finalization

### PR DESCRIPTION
**Pull request by trianglesphere**
**Description**

Fix the finalization description. See also the design doc for how it should work: https://github.com/ethereum-optimism/design-docs/blob/main/protocol/plasma-mode.md#finalization

### CODE REVIEW

File: specs/experimental/plasma.md [hunk 1]
```diff
@@ -281,17 +281,14 @@ and users must use the “finalized” label for a guarantee that their state ca
...
-    l1_finalized_block_number = min(latest_l1_block_number - finality_delay, finalized_l1_block_number)
-else 
-    l1_finalized_block_number = min(latest_l1_block_number - challenge_window, finalized_l1_block_number)
-```
+as they are validated by the derivation pipeline.
+
+The L2 chain can be marked as finalized when the L1 block in which commitments can no longer be invalidated
+becomes finalized. Previously L2 blocks are fully derived from an L1 block when the batch is fully available.
+Plasma commitments are only partially derived when the commitment is found in a L1 block ("plasma safe"), and
+the become fully derived when that commitment can no longer be invalidated ("safe"). Then finalization can
```
Comment (ajsutton) [inline specs/experimental/plasma.md]:
```suggestion
then become fully derived when that commitment can no longer be invalidated ("safe"). Then finalization can
```

File: specs/experimental/plasma.md [hunk 2]
```diff
@@ -281,17 +281,14 @@ and users must use the “finalized” label for a guarantee that their state ca
...
+proceed as normal: when the L1 block than an L2 block is fully derived from becomes finalized, the L2 block
```
Comment (ajsutton) [inline specs/experimental/plasma.md]:
```suggestion
proceed as normal: when the L1 block that an L2 block is fully derived from becomes finalized, the L2 block
```

File: specs/experimental/plasma.md [hunk 3]
```diff
@@ -281,17 +281,14 @@ and users must use the “finalized” label for a guarantee that their state ca
...
-    finality_delay = (challenge.block_number - challenge.commitment_block_number) + resolve_window + 1
-    l1_finalized_block_number = min(latest_l1_block_number - finality_delay, finalized_l1_block_number)
-else 
-    l1_finalized_block_number = min(latest_l1_block_number - challenge_window, finalized_l1_block_number)
-```
+as they are validated by the derivation pipeline.
+
+The L2 chain can be marked as finalized when the L1 block in which commitments can no longer be invalidated
+becomes finalized. Previously L2 blocks are fully derived from an L1 block when the batch is fully available.
+Plasma commitments are only partially derived when the commitment is found in a L1 block ("plasma safe"), and
```
Comment (ajsutton) [inline specs/experimental/plasma.md]: Is partially derived the right term here or would it be better to say they're optimistically derived? ie the full block and state is derived, it just hasn't completed all the validation checks yet so is still optimistic.

**Review by ajsutton — APPROVED** [review]
Couple of typos and a possible wording change but feel free to ignore that.

File: specs/experimental/plasma.md (hunk 3 above)
Comment (trianglesphere) [reply specs/experimental/plasma.md]: Yea, optimistically derived is good term. Realistically we need to introduce "plasma safe" and we're marking blocks as actual "safe" too early right now.

Comment (trianglesphere) [reply specs/experimental/plasma.md]: Though I guess usually if we have the data, we should be able to win any challenge made against the commitment so it's pretty close to safe if we are also running a challenger agent + have stored the data somewhere.

**Pull request merged**